packages/calendar/components/ChannelFilter/index.jsx

```diff
@@ -25,14 +32,35 @@ const ChannelFilter = ({ channelFilter, channelSelected }) => {
   }`;
   const label = allChannelsAreSelected ? 'All channels' : labelForSoloChannels;
 
+  const handleFilteredChannels = ({ beforeFilterCount, afterFilterCount }) => {
+    trackChannelsFiltered({
+      isWeekly,
+      beforeFilterCount,
+      afterFilterCount,
+      isViewOnly,
+    });
+  };
+
+  const handleSelectedChannels = ({ beforeFilterCount, afterFilterCount }) => {
+    trackChannelsSelected({ isWeekly, beforeFilterCount, afterFilterCount });
+  };
+
   // Select default filter values
   useEffect(() => {
     // Filter by one channel if coming from route with channel state (e.g. queue)
     if (channelSelected) {
       setChannels([channelSelected]);
+      handleFilteredChannels({
+        beforeFilterCount: channels.length,
+        afterFilterCount: 1,
+      });
       // Or select all channels
-    } else if (numberOfChannelsSelected === 0) {
+    } else if (numberOfChannelsSelected === 0 && !loading) {
       onSelectAllClick({ setChannels, channels });
+      handleFilteredChannels({
+        beforeFilterCount: numberOfChannelsSelected,
```

Comment:
Since this only runs when there are no channelsSelected,
```suggestion
        beforeFilterCount: 0,
```

Reply:
Makes sense, Ana! Thanks : )